### ISSUE TITLE
Clarification about epochs and batch size

### ISSUE THREAD

**Issue by david-cortes** [lenskit/lkpy#256]
In `IntegratedBiasMF`, there are parameters `epochs` and `batch_size`. As I understand it, an epoch should constitute a full pass over the data, so that if e.g. there are 100 observations and a batch size of 25, it would do 4 steps per epoch.

However - and I might be wrong about this - when looking at the training loss that is printed, I see that it prints "epochs" with a size matching to the batch size, thus counting each update as an epoch.

**Comment by mdekstrand** [comment]
Yes, that is a correct interpretation; epochs and batch size are both passed directly to TensorFlow, however, so their precise interpretation - and the progress output - are controlled by TF, not LensKit.

Could you provide an example of the output you're seeing?

**Comment by david-cortes** [comment]
I now realize that the number shown in the training loss is the number of steps per epoch. Tried with e.g. 1000 observations and batch size of 50, it shows 20 steps, with 25 it shows 40, and so on.

Is that the correction behavior? I thought keras was supposed to show the total number of observations in the training logs.

Example:
```python
import numpy as np, pandas as pd
from lenskit.algorithms.tf import IntegratedBiasMF

nusers = int(1e2)
nitems = int(1e2)
nnz = int(1e3)

np.random.seed(1)
df = pd.DataFrame({
    "user" : np.random.randint(nusers, size=nnz),
    "item" : np.random.randint(nitems, size=nnz),
    "rating" : np.random.gamma(1,1, size=nnz)
})

model = IntegratedBiasMF(features=5,
                         epochs=1,
                         batch_size=25)
model.fit(df)
```
```
40/40 [==============================] - 1s 12ms/step - loss: 1.1835
```

**Comment by mdekstrand** [comment]
I haven't spent enough time with Keras to have a full handle on how it does its logging (this is the first thing I built with Keras, actually), but for IntegratedBiasMF, `batch_size` and `epochs` are passed directly to Keras `model.fit`. (This is not true for the BPR model, as it uses a custom data set implementation for the negative sampling.)

**Comment by david-cortes** [comment]
Did a bit more digging into keras and I think it's working as expected.

**Comment by mdekstrand** [comment]
@david-cortes thank you for the follow-up!
